### PR TITLE
Codecs package docs, as well as some assorted tweaks or minor additions

### PR DESCRIPTION
**Pull request by harjitmoe**
Mostly this involved adding a large number of docstrings to the codecs package, and adjusting the documentation generator and CSS to better accommodate the codecs package (improving the readability of tables, and linking from parent packages to child modules and _vice versa_ for navigation).

Additional changes made at the same time:

- `encode_jis7_reduced` was moved out of `dbextra` into the expected `dbextra_data_7bit`, allowing it to be lazy evaluated.
- `Cp874IncrementalDecoder` and `Cp874IncrementalEncoder` were removed as redundant to `Windows874IncrementalDecoder` and `Windows874IncrementalEncoder`, and their labels transferred (the former differed in not mapping the non-repurposed C1 codes, but that hardly warrants being included when the latter already is).&ensp;This redundancy had passed unnoticed while I was adding single-byte mappings for Python parity.
- An additional criterion allowing narrowing down to ISO-2022-JP was added to `JapaneseAutodetectIncrementalDecoder` (this mainly improves handling of incremental use with small segments per call which will not, or are unlikely to, contain an entire escape sequence in a single segment, but are likely to not switch the decoder state back to ASCII by the end of the segment).
- Several instances of code referencing a non-existent local, which would never execute given the current set of error handlers, but would if someone registered an error handler which returned a negative resumption index, were fixed.
- A `cp042` codec was added to handle [code page 42](http://archives.miloush.net/michkap/archive/2005/11/08/490495.html) sometimes used by Windows to map symbol fonts to the Private Use Area (this apparently maps to an entirely different range than `x-user-defined`, despite being very similar in concept besides not being Extended ASCII).

### CODE REVIEW

File: docs/doxy.css [hunk 1]
```diff
@@ -388,3 +388,14 @@ div.memdoc {
   color: #bbb;
   content: '->';
 }
+
+.markdownTableRowOdd {
+    background: rgba(204, 204, 204, 0.2);
+}
+table {
+    margin-bottom: 1em;
```
Comment (klange) [inline docs/doxy.css:396]: Doxygen throws tables around like it's still the early 2000s; can you check how this affects the spacing on the C API function/member listings?

File: .gitignore [hunk 2]
```diff
@@ -24,4 +24,6 @@
 /modules/codecs/sbencs.krk
 /modules/codecs/dbdata.krk
 /modules/codecs/isweblabel.krk
+#
+.fuse_hidden*
```
Comment (klange) [inline .gitignore:28]: Please move this to your user `.gitignore`.

File: tools/gendoc.krk [hunk 3]
```diff
@@ -279,6 +288,23 @@ def processModules(modules):
                     else:
                         other.append(Pair(member,obj))
 
+            # There will probably ultimately be a better way of doing this
+            let isPackage = False
```
Comment (klange) [inline tools/gendoc.krk:292]: `module.__ispackage__` 

(ref: https://github.com/kuroko-lang/kuroko/blob/master/src/vm.c#L1592)

File: docs/doxy.css (hunk 1 above)
Comment (harjitmoe) [reply docs/doxy.css:396]: I've changed that now to constrain the selectors further with the `.markdownTable` class, which should exclude the interface tables generated by Doxygen itself.

File: .gitignore (hunk 2 above)
Comment (harjitmoe) [reply .gitignore]: Done.

File: tools/gendoc.krk (hunk 3 above)
Comment (harjitmoe) [reply tools/gendoc.krk]: Changed to `hasattr(module, "__ispackage__") and module.__ispackage__` (I initially tried just `module.__ispackage__`, but it didn't seem to exist on non-packages).

**Pull request merged**